### PR TITLE
Initial specs for browser extension

### PR DESCRIPTION
**Pull request by jurevans**
Defining specifications for Anoma browser extension (key management, account creation, API for signing transactions via connected web applications, login session management)

Relates to:

- https://github.com/anoma/namada-interface/issues/39
- https://github.com/anoma/namada-interface/issues/41

__TODO__ The documentation for the `web-wallet` should also be updated to reflect the changes here. The extension serves as the "wallet", so perhaps we should move `web-wallet` to `namada-interface` to be more clear about the intention of each app

### CODE REVIEW

File: src/architecture/clients/browser-extension/architecture.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+The proxy is necessary to forward data via messages (using `window.postMessage`) to execute functions within the extension, as well as return message data to the client.
+
+Essentially, any wasm- or storage-related functionality will be handled by the Service Worker (or background script in Firefox). This background process has access to IndexedDB,
+and is configured in the manifest versions 2 and 3 to execute wasm code.
+
+## Authentication
+
+In `@keplr-wallet/extension`, authentication is handled by retrieving an encrypted mnemonic phrase and attempting to decrypt it, which if successful, will set the provided
+password in the state of the background-scripts class instance for some duration. This is how authenication works on `namada-interface`, with the caveat that being a client-side-only
```
Comment (memasdeligeorgakis) [inline src/architecture/clients/browser-extension/architecture.md:53]:
```suggestion
password in the state of the background-scripts class instance for some duration. This is how authentication works on `namada-interface`, with the caveat that being a client-side-only
```

File: src/architecture/clients/browser-extension/interface.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+to reuse similar components and development processes as we currently employ.
+
+The stack should resemble the following:
+
+- TypeScript
+- React
+- Redux-Toolkit (`@reduxjs/toolkit`)
+- Styled-Components
+
+While this stack is generally consistent, we should perhaps move away from `create-react-app`, as this imits the control we
```
Comment (memasdeligeorgakis) [inline src/architecture/clients/browser-extension/interface.md:13]:
```suggestion
While this stack is generally consistent, we should perhaps move away from `create-react-app`, as this limits the control we

File: src/architecture/clients/browser-extension/architecture.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+The flow may resemble the following:
+
+1. The interface accepts a password and posts a message to authenticate via the proxy
+2. The proxy fetches the decrypted mnemonic from storage and attempts to decrypt it
+3. If successful, the users credentials are proxied back to the extension for persisting their session
+
+## Manifest considerations
+
+Currently, Firefox does not support `manifest_version: 3`, but Chrome no longer accepts `manifest_version: 2` into their extension store. It is best that we follow what the `metamask-extension`
+is doing by supporting multiple both versions. `@keplr-wallet/extension` is currently using only version 2 in all browsers, but as we will be a new submission, we need to support version 3.
```
Comment (memasdeligeorgakis) [inline src/architecture/clients/browser-extension/architecture.md:66]:
```suggestion
is doing by supporting both versions. `@keplr-wallet/extension` is currently using only version 2 in all browsers, but as we will be a new submission, we need to support version 3.
```

**Comment by jurevans** [comment]
Closing this in favor of: https://github.com/anoma/namada-interface/pull/88

**Pull request closed without merging**